cl/citations/tests.py

```diff
@@ -355,39 +429,350 @@ def test_disambiguate_citations(self):
+        test_pairs = [
+            # Simple example for full citations
+            ('asdf 22 U.S. 33 asdf', full_citation_html),
+
+            # Using a variant format for U.S. (Issue #409)
+            ('asdf 22 U. S. 33 asdf', full_citation_html),
+
+            # Full citation across line break
+            ('asdf John v. Doe, 123\nU.S. 456, upholding foo bar',
+             '<pre class="inline">asdf John v. Doe, </pre><span class="'
+             'citation no-link"><span class="volume">123</span>\n<span class='
+             '"reporter">U.S.</span> <span class="page">456</span></span><pre'
+             ' class="inline">, upholding foo bar</pre>'),
+
+            # First kind of short form citation (meaningless antecedent)
+            ('asdf. 515 U.S., at 240. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf. </span><span class="volume">515</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">240'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Second kind of short form citation (meaningful antecedent)
+            ('asdf, 1 U. S., at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Short form citation with no comma after reporter in original
+            ('asdf, 1 U. S. at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Short form citation across line break
+            (u'asdf.’ ” 123 \n U.S., at 456. Foo bar foobar',
+             '<pre class="inline">asdf.’ </pre><span class="citation no-link">'
+             '<span class="antecedent">” </span><span class="volume">123'
+             '</span> \n <span class="reporter">U.S.</span>, at <span class='
+             '"page">456</span></span><pre class="inline">. Foo bar foobar'
+             '</pre>'),
+
+            # First kind of supra citation (standard kind)
+            ('asdf, supra, at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span><span> supra</span><span>, at '
+             '</span><span class="page">2</span></span><pre class="inline">'
+             '. foobar</pre>'),
+
+            # Second kind of supra citation (with volume)
+            ('asdf, 123 supra, at 2. foo bar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
```

Comment:
Weird to have a pre opening and closing itself here, with nothing inside?

Reply:
Yes, not sure what is happening, I will fix this

Reply:
It turns out that this behavior is intended, but just looks weird because the test string is too short. I added some more words at the beginning to make the test more realistic and to demonstrate what the full generated HTML is. b1e76cd

Reply:
Ah ha.